File: 다음큰숫자/java/20221223/src/Solution.java
public class Solution {
    public int countOne(int number) {
        return Integer.bitCount(number);
    }

    public int findNextBigger(int number, int ones) {
        if (countOne(number + 1) == ones) {
            return number + 1;
        }

        return findNextBigger(number + 1, ones);
    }

    public int solution(int number) {
        return findNextBigger(number, countOne(number));
    }
}

File: 다음큰숫자/javascript/20221223/solution.js
export const countOne = (number) => (number.toString(2)
  .split('')
  .reduce((acc, current) => (current === "1" ? acc + 1 : acc), 0))


export const findNextBigger = (number) => {
  const toMatch = countOne(number);

  for (let i = number + 1; ; i += 1) {
    if (countOne(i) === toMatch) {

      return i;
    }
  }
}
